crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -676,40 +678,64 @@ where
                 value,
                 ..
             }) => {
-                // If we're in a class or module scope, then the annotation needs to be
-                // available at runtime.
-                // See: https://docs.python.org/3/reference/simple_stmts.html#annotated-assignment-statements
-                let runtime_annotation = if self.semantic.future_annotations() {
-                    self.semantic
+                enum AnnotationKind {
+                    RuntimeRequired,
+                    RuntimeEvaluated,
+                    TypingOnly,
+                }
+
+                fn annotation_kind(
+                    semantic: &SemanticModel,
+                    settings: &LinterSettings,
+                ) -> AnnotationKind {
+                    // If the annotation is in a class, and that class is marked as
```

Comment:
Here, with "if the annotation is in a class" we're specifically referring to the class scope i.e.,
```python
class Foo:
	bar: str
	#    ^^^
```
Is that correct? If so, it would be worth updating the comment to reflect that.